packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol

```diff
@@ -412,7 +413,13 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
             // Ignore false claims
             if (claim.counteredBy == address(0)) {
                 countered = claim.claimant;
-                break;
+                leftmostCounter = claim.position;
+            } else {
+                // These claims are at the same depth, so we can compare them using their gindices
+                if (leftmostCounter.raw() > claim.position.raw()) {
+                    countered = claim.claimant;
+                    leftmostCounter = claim.position;
+                }
```

Comment:
I don't think this logic is correct. "If subgame is countered, set the `countered` and `leftmostCounter` flag for the parent. Else if the current leftmost counter is further right than the claim's position, set the the `countered` and `leftmostCounter` flag for the parent." What if the claim that's further left is not uncountered?

What do you think about pairing the check?

```solidity
// Assume parent is honest until proven otherwise
address countered = address(0);
Position leftmostCounter = Position.wrap(type(uint128).max);
for (uint256 i = 0; i < challengeIndicesLen; ++i) {
    uint256 challengeIndex = challengeIndices[i];

    // INVARIANT: Cannot resolve a subgame containing an unresolved claim
    if (subgames[challengeIndex].length != 0) revert OutOfOrderResolution();

    ClaimData storage claim = claimData[challengeIndex];

    // If the child subgame is uncountered and further left than the current left most counter,
    // update the parent subgame's `countered` address and the current `leftmostCounter`.
    // The left-most correct counter is preferred in bond payouts due to possible "poison claims,"
    // where an attacker can defend an incorrect claim with the correct claim hash, when they should
    // have attacked.
    if (claim.counteredBy == address(0) && leftmostCounter.raw() > claim.position.raw()) {
        countered = claim.claimant;
        leftmostCounter = claim.position;
    }
}
```

Reply:
Good catch. I misread `claim.counteredBy` as `countered` not set. Need to update tests to catch this too.